### PR TITLE
[Navigation] Add white background to Navigation if logo is present

### PR DESCRIPTION
**Pull request by mateus**
### WHY are these changes introduced?

Adding a white background color to the logo container area of the Navigation component. 

Screen Shot 2022-06-15 at 3 33 28 PM

### WHAT is this pull request doing?

Add a white background color and border bottom to the Navigation logo container. This is the same approach as the one found on the TopBar component. In case the `contextControl` prop is provided, these style changes will not be applied. 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {
  HomeMinor,
  OrdersMinor,
  MarketingMinor,
  ProductsMinor,
} from '@shopify/polaris-icons';

import {Navigation, Frame, Button} from '../src';

export function Playground() {
  const logo = {
    width: 100,
    topBarSource: `data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAwIiBoZWlnaHQ9IjMwIiBmaWxsPSJub25lIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPjxwYXRoIGQ9Ik0yMS40OCA2LjEwMWEuMjUzLjI1MyAwIDAgMC0uMjQtLjIyYy0uMSAwLTIuMDgtLjA0LTIuMDgtLjA0cy0xLjY2LTEuNi0xLjgyLTEuNzhjLS4xNi0uMTYtLjQ4LS4xMi0uNi0uMDhsLS44NC4yNmMtLjA4LS4yOC0uMjItLjYyLS40LS45OC0uNTgtMS4xMi0xLjQ2LTEuNzItMi41LTEuNzItLjA4IDAtLjE0IDAtLjIyLjAyLS4wNC0uMDQtLjA2LS4wOC0uMS0uMS0uNDYtLjQ4LTEuMDQtLjcyLTEuNzQtLjctMS4zNC4wNC0yLjY4IDEuMDItMy43OCAyLjc0LS43NiAxLjIyLTEuMzQgMi43NC0xLjUyIDMuOTItMS41NC40OC0yLjYyLjgyLTIuNjYuODItLjc4LjI0LS44LjI2LS45IDEtLjA0LjU2LTIuMDggMTYuMzQtMi4wOCAxNi4zNGwxNy4xMiAyLjk2IDcuNDItMS44NGMtLjAyIDAtMy4wNC0yMC40Ni0zLjA2LTIwLjZabS02LjQ0LTEuNThjLS40LjEyLS44NC4yNi0xLjMyLjQyIDAtLjY4LS4xLTEuNjQtLjQtMi40NCAxLjAyLjE2IDEuNTIgMS4zMiAxLjcyIDIuMDJabS0yLjIyLjY4Yy0uOS4yOC0xLjg4LjU4LTIuODYuODguMjgtMS4wNi44LTIuMSAxLjQ0LTIuOC4yNC0uMjYuNTgtLjU0Ljk2LS43LjQuNzguNDggMS44OC40NiAyLjYyWm0tMS44NC0zLjU0Yy4zMiAwIC41OC4wNi44LjIyLS4zNi4xOC0uNzIuNDYtMS4wNC44Mi0uODYuOTItMS41MiAyLjM0LTEuNzggMy43Mi0uODIuMjYtMS42Mi41LTIuMzQuNzIuNDgtMi4xOCAyLjMtNS40MiA0LjM2LTUuNDhaIiBmaWxsPSIjOTVCRjQ3Ii8+PHBhdGggZD0iTTIxLjI0IDUuODgxYy0uMSAwLTIuMDgtLjA0LTIuMDgtLjA0cy0xLjY2LTEuNi0xLjgyLTEuNzhhLjMyMS4zMjEgMCAwIDAtLjIyLS4xdjI0LjU4bDcuNDItMS44NC0zLjA0LTIwLjZjLS4wNC0uMTQtLjE2LS4yMi0uMjYtLjIyWiIgZmlsbD0iIzVFOEUzRSIvPjxwYXRoIGQ9Im0xMyA5LjcwMS0uODYgMy4yMnMtLjk2LS40NC0yLjEtLjM2Yy0xLjY4LjEtMS42OCAxLjE2LTEuNjggMS40Mi4xIDEuNDQgMy44OCAxLjc2IDQuMSA1LjE0LjE2IDIuNjYtMS40IDQuNDgtMy42OCA0LjYyLTIuNzIuMTQtNC4yMi0xLjQ2LTQuMjItMS40NmwuNTgtMi40NnMxLjUyIDEuMTQgMi43MiAxLjA2Yy43OC0uMDQgMS4wOC0uNyAxLjA0LTEuMTQtLjEyLTEuODgtMy4yLTEuNzYtMy40LTQuODYtLjE2LTIuNiAxLjU0LTUuMjIgNS4zLTUuNDYgMS40Ni0uMSAyLjIuMjggMi4yLjI4WiIgZmlsbD0iI2ZmZiIvPjxwYXRoIGQ9Ik0zNC41OCAxNi41NjFjLS44Ni0uNDYtMS4zLS44Ni0xLjMtMS40IDAtLjY4LjYyLTEuMTIgMS41OC0xLjEyIDEuMTIgMCAyLjEyLjQ2IDIuMTIuNDZsLjc4LTIuNHMtLjcyLS41Ni0yLjg0LS41NmMtMi45NiAwLTUuMDIgMS43LTUuMDIgNC4wOCAwIDEuMzYuOTYgMi4zOCAyLjI0IDMuMTIgMS4wNC41OCAxLjQgMSAxLjQgMS42MiAwIC42NC0uNTIgMS4xNi0xLjQ4IDEuMTYtMS40MiAwLTIuNzgtLjc0LTIuNzgtLjc0bC0uODQgMi40czEuMjQuODQgMy4zNC44NGMzLjA0IDAgNS4yNC0xLjUgNS4yNC00LjItLjAyLTEuNDYtMS4xMi0yLjUtMi40NC0zLjI2Wk00Ni43IDExLjUwMWMtMS41IDAtMi42OC43Mi0zLjU4IDEuOGwtLjA0LS4wMiAxLjMtNi44SDQxbC0zLjMgMTcuMzJoMy4zOGwxLjEyLTUuOTJjLjQ0LTIuMjQgMS42LTMuNjIgMi42OC0zLjYyLjc2IDAgMS4wNi41MiAxLjA2IDEuMjYgMCAuNDYtLjA0IDEuMDQtLjE0IDEuNWwtMS4yOCA2Ljc4aDMuMzhsMS4zMi03Yy4xNC0uNzQuMjQtMS42Mi4yNC0yLjIyLjAyLTEuOTItLjk4LTMuMDgtMi43Ni0zLjA4Wk01Ny4xNCAxMS41MDFjLTQuMDggMC02Ljc4IDMuNjgtNi43OCA3Ljc4IDAgMi42MiAxLjYyIDQuNzQgNC42NiA0Ljc0IDQgMCA2LjctMy41OCA2LjctNy43OC4wMi0yLjQyLTEuNC00Ljc0LTQuNTgtNC43NFptLTEuNjYgOS45NGMtMS4xNiAwLTEuNjQtLjk4LTEuNjQtMi4yMiAwLTEuOTQgMS01LjEgMi44NC01LjEgMS4yIDAgMS42IDEuMDQgMS42IDIuMDQgMCAyLjA4LTEuMDIgNS4yOC0yLjggNS4yOFpNNzAuNCAxMS41MDFjLTIuMjggMC0zLjU4IDIuMDItMy41OCAyLjAyaC0uMDRsLjItMS44MmgtM2MtLjE0IDEuMjItLjQyIDMuMS0uNjggNC41bC0yLjM2IDEyLjRoMy4zOGwuOTQtNS4wMmguMDhzLjcuNDQgMS45OC40NGMzLjk4IDAgNi41OC00LjA4IDYuNTgtOC4yIDAtMi4yOC0xLjAyLTQuMzItMy41LTQuMzJabS0zLjI0IDkuOThjLS44OCAwLTEuNC0uNS0xLjQtLjVsLjU2LTMuMTZjLjQtMi4xMiAxLjUtMy41MiAyLjY4LTMuNTIgMS4wNCAwIDEuMzYuOTYgMS4zNiAxLjg2IDAgMi4yLTEuMyA1LjMyLTMuMiA1LjMyWk03OC43NCA2LjY0MWMtMS4wOCAwLTEuOTQuODYtMS45NCAxLjk2IDAgMSAuNjQgMS43IDEuNiAxLjdoLjA0YzEuMDYgMCAxLjk2LS43MiAxLjk4LTEuOTYgMC0uOTgtLjY2LTEuNy0xLjY4LTEuN1pNNzQgMjMuNzgxaDMuMzhsMi4zLTEyaC0zLjRsLTIuMjggMTJaTTg4LjMgMTEuNzYxaC0yLjM2bC4xMi0uNTZjLjItMS4xNi44OC0yLjE4IDIuMDItMi4xOC42IDAgMS4wOC4xOCAxLjA4LjE4bC42Ni0yLjY2cy0uNTgtLjMtMS44NC0uM2MtMS4yIDAtMi40LjM0LTMuMzIgMS4xMi0xLjE2Ljk4LTEuNyAyLjQtMS45NiAzLjg0bC0uMS41NmgtMS41OGwtLjUgMi41NmgxLjU4bC0xLjggOS40OGgzLjM4bDEuOC05LjQ4aDIuMzRsLjQ4LTIuNTZaTTk2LjQ2IDExLjc4MXMtMi4xMiA1LjM0LTMuMDYgOC4yNmgtLjA0Yy0uMDYtLjk0LS44NC04LjI2LS44NC04LjI2aC0zLjU2TDkxIDIyLjgwMWMuMDQuMjQuMDIuNC0uMDguNTYtLjQuNzYtMS4wNiAxLjUtMS44NCAyLjA0LS42NC40Ni0xLjM2Ljc2LTEuOTIuOTZsLjk0IDIuODhjLjY4LS4xNCAyLjEyLS43MiAzLjMyLTEuODQgMS41NC0xLjQ0IDIuOTgtMy42OCA0LjQ0LTYuNzJsNC4xNC04LjloLTMuNTRaIiBmaWxsPSIjMDAwIi8+PC9zdmc+`,
    url: 'https://shopify.com',
    accessibilityLabel: 'Shopify ',
  };

  const navigation = (
    <Navigation location="/">
      {/* <div style={{margin: '0 16px 16px'}}>
        <Button outline fullWidth textAlign="left">
          Shop 1
        </Button>
      </div> */}
      <Navigation.Section
        items={[
          {
            url: '/path/to/place',
            label: 'Home',
            icon: HomeMinor,
            selected: true,
          },
          {
            url: '/path/to/place',
            label: 'Orders',
            icon: OrdersMinor,
            badge: '15',

            subNavigationItems: [
              {
                url: '/admin/orders/collections',
                disabled: false,
                selected: false,
                label: 'Collections',
              },
              {
                url: '/admin/orders/inventory',
                disabled: false,
                label: 'Inventory',
              },
            ],
          },
          {
            url: '/path/to/place',
            label: 'Marketing',
            icon: MarketingMinor,
            badge: '15',
            subNavigationItems: [
              {
                url: '/admin/analytics/collections',
                disabled: false,
                selected: false,
                label: 'Reports',
              },
              {
                url: '/admin/analytics/inventory',
                disabled: false,
                label: 'Live view',
              },
            ],
          },
          {
            url: '/admin/products',
            label: 'Products',
            icon: ProductsMinor,
            selected: true,
            subNavigationItems: [
              {
                url: '/?path=/story/all-components-navigation--navigation-with-multiple-secondary-navigations',
                disabled: false,
                selected: false,
                label: 'Collections',
              },
              {
                url: '/admin/products/inventory',
                disabled: false,
                selected: true,
                label: 'Inventory',
              },
            ],
          },
        ]}
      />
    </Navigation>
  );

  return (
    <div style={{height: '250px'}}>
      <Frame logo={logo} navigation={navigation} showMobileNavigation />
    </div>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-react/src/components/Navigation/Navigation.scss [hunk 1]
```diff
@@ -71,6 +72,18 @@ $nav-max-width: 360px;
...
+    box-shadow: var(--p-shadow-top-bar);
+    margin-bottom: var(--p-space-4);
+
+    &::after {
+      content: '';
+      position: absolute;
+      bottom: 0;
+      width: 100%;
+      border-bottom: var(--p-border-width-1) solid transparent;
+    }
```
Comment (alex-page) [inline polaris-react/src/components/Navigation/Navigation.scss]: Why does the `border-bottom` need to be positioned like this? Could we move this to a parent element if it's trying the break a bounding area? Does the border-bottom work after the `margin-bottom`?

Comment (mateus) [reply polaris-react/src/components/Navigation/Navigation.scss]: That was an oversight. I removed it.

**Review by alex-page — APPROVED** [review]
This looks fine to me. Has this been tophatted in global nav where they override the colors?

**Comment by mateus** [comment]
> This looks fine to me. Has this been tophatted in global nav where they override the colors?

I am testing it with the snapshot now.

**Review by kyledurand — APPROVED** [review]

**Comment by mateus** [comment]
> > This looks fine to me. Has this been tophatted in global nav where they override the colors?
> 
> I am testing it with the snapshot now.

Screen Shot 2022-06-15 at 6 10 05 PM

Since it's using the contextControl prop the `LogoContainer` div does not get rendered.  👍🏻

File: .changeset/stupid-books-unite.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': minor
+---
+
+Add white background to Navigation if logo is present
```
Comment (chloerice) [inline .changeset/stupid-books-unite.md]:
```suggestion
Updated the `Navigation` `logo` container to have a white background when present
```

**Review by chloerice — APPROVED** [review]
🚀 

(Don't worry about the size limit check, it's being fixed in #6137)

**Pull request merged**